### ISSUE TITLE
Question in code/model/agent.py

### ISSUE THREAD

**Issue by liujshi** [shehzaadzd/MINERVA#4]
```
action = tf.to_int32(tf.multinomial(logits=scores, num_samples=1))  # [B, 1]

# loss
# 5a.
label_action =  tf.squeeze(action, axis=1)
loss = tf.nn.sparse_softmax_cross_entropy_with_logits(logits=scores, labels=label_action)  # [B,]
```
the above code confused me, action is sampled from scores, and loss is cross entropy of action and scores, what's the meaning of the loss?

thank you!

**Comment by rajarshd** [comment]
Hi @liujshi ,
That is used to compute the gradient of the expected reward as given by the REINFORCE (Williams, 92) algorithm. 
![screen shot 2018-03-10 at 1 10 03 pm](https://user-images.githubusercontent.com/2314699/37245340-7b6fe490-2464-11e8-90dc-4cb79d97acda.png)

Hope this is clear now.

**Comment by zhujiagang** [comment]
```
        # Masking PAD actions
        comparison_tensor = tf.ones_like(next_relations, dtype=tf.int32) * self.rPAD  # matrix to compare
        mask = tf.equal(next_relations, comparison_tensor)  # The mask
        dummy_scores = tf.ones_like(prelim_scores) * -99999.0  # the base matrix to choose from if dummy relation
        scores = tf.where(mask, dummy_scores, prelim_scores)  # [B, MAX_NUM_ACTIONS]
```

Is this the code segment that realizes sampling action only from outgoing edges as said in your paper?
